ui/app/actions.js

```diff
@@ -746,20 +745,26 @@ function updateGasData ({
 }) {
   return (dispatch) => {
     dispatch(actions.gasLoadingStarted())
-    const estimatedGasPrice = estimateGasPriceFromRecentBlocks(recentBlocks)
-    return Promise.all([
-      Promise.resolve(estimatedGasPrice),
-      estimateGas({
+    let gasPrice
+    return new Promise((resolve, reject) => {
+       background.getGasPrice((err, data) => {
```

Comment:
Looks like the indentation is inconsistent.

Reply:
We need the value resolved by the first promise, it then needs to be passed to `estimateGas` which returns another promise, and then is needed again after `estimateGas` resolves.